core/src/test/java/kafka/test/api/AdminClientRebootstrapTest.java

```diff
@@ -0,0 +1,111 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.test.api;
+
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.common.config.TopicConfig;
+import org.apache.kafka.common.test.ClusterInstance;
+import org.apache.kafka.common.test.api.ClusterConfig;
+import org.apache.kafka.common.test.api.ClusterTemplate;
+import org.apache.kafka.common.test.api.Type;
+import org.apache.kafka.coordinator.group.GroupCoordinatorConfig;
+import org.apache.kafka.test.TestUtils;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+public class AdminClientRebootstrapTest {
+    private static final int BROKER_COUNT = 2;
+
+    private static List<ClusterConfig> generator() {
+        // Enable unclean leader election for the test topic
+        Map<String, String> serverProperties = Map.of(
+            TopicConfig.UNCLEAN_LEADER_ELECTION_ENABLE_CONFIG, "true",
+            GroupCoordinatorConfig.OFFSETS_TOPIC_REPLICATION_FACTOR_CONFIG, String.valueOf(BROKER_COUNT)
+        );
+
+        return Stream.of(false, true)
+            .map(AdminClientRebootstrapTest::getRebootstrapConfig)
+            .map(rebootstrapProperties -> AdminClientRebootstrapTest.buildConfig(serverProperties, rebootstrapProperties))
+            .toList();
+    }
+
+    private static Map<String, String> getRebootstrapConfig(boolean useRebootstrapTriggerMs) {
+        Map<String, String> properties = new HashMap<>();
+        if (useRebootstrapTriggerMs) {
+            properties.put(CommonClientConfigs.METADATA_RECOVERY_REBOOTSTRAP_TRIGGER_MS_CONFIG, "5000");
+        } else {
+            properties.put(CommonClientConfigs.METADATA_RECOVERY_REBOOTSTRAP_TRIGGER_MS_CONFIG, "3600000");
+            properties.put(CommonClientConfigs.SOCKET_CONNECTION_SETUP_TIMEOUT_MS_CONFIG, "5000");
+            properties.put(CommonClientConfigs.SOCKET_CONNECTION_SETUP_TIMEOUT_MAX_MS_CONFIG, "5000");
+            properties.put(CommonClientConfigs.RECONNECT_BACKOFF_MS_CONFIG, "1000");
+            properties.put(CommonClientConfigs.RECONNECT_BACKOFF_MAX_MS_CONFIG, "1000");
+        }
+        properties.put(CommonClientConfigs.METADATA_RECOVERY_STRATEGY_CONFIG, "rebootstrap");
+        return properties;
+    }
+
+    private static ClusterConfig buildConfig(Map<String, String> serverProperties, Map<String, String> rebootstrapProperties) {
+        return ClusterConfig.defaultBuilder()
+            .setTypes(Set.of(Type.KRAFT))
+            .setBrokers(BROKER_COUNT)
+            .setAdminClientProperties(rebootstrapProperties)
```

Comment:
this method is no-op, and we should use `clusterInstance.admin(Map)` to build admin with custom configs.

Reply:
open https://issues.apache.org/jira/browse/KAFKA-18944 to remove those unused setters